server/odc-service/src/main/java/com/oceanbase/odc/service/task/runtime/RollbackPlanTask.java

```diff
@@ -0,0 +1,276 @@
+                        break;
+                    }
+                    String sql;
+                    if (CollectionUtils.isNotEmpty(userInputSqls)) {
+                        sql = userInputSqls.remove(0).getStr();
+                    } else {
+                        sql = uploadFileSqlIterator.next().getStr();
+                    }
+                    try {
+                        long timeoutForCurrentSql = timeoutMills - (System.currentTimeMillis() - startTimeMills);
+                        if (timeoutForCurrentSql <= 0) {
+                            log.warn("Generate rollback plan task has timeout, timeout milliseconds={}, taskId={}",
+                                    timeoutMills, taskId);
+                            rollbackPlans.append("/* Generate rollback plan task has timeout, timeout milliseconds=")
+                                    .append(timeoutMills).append(", generate rollback plan will be stopped. */\n");
+                            handleRollbackResult(rollbackPlans.toString());
+                            this.success = true;
+                            return;
+                        }
+                        GenerateRollbackPlan rollbackPlan = RollbackGeneratorFactory.create(sql,
+                                parameters.getRollbackProperties(), session, timeoutForCurrentSql);
+                        RollbackPlan result = rollbackPlan.generate();
+                        totalChangeLines += result.getChangeLineCount();
+                        rollbackPlans.append(result);
+                        if (totalChangeLines > totalMaxChangeLines) {
+                            log.info(
+                                    "The number of changed lines for taskId={} exceeds the maximum limit, changed line count={}",
+                                    taskId, totalChangeLines);
+                            rollbackPlans.append("/* The number of changed lines exceeds the maximum limit:")
+                                    .append(totalMaxChangeLines)
+                                    .append(", generate rollback plan will be stopped. */\n");
+                            handleRollbackResult(rollbackPlans.toString());
+                            this.success = true;
+                            return;
+                        }
+                    } catch (UnsupportedSqlTypeForRollbackPlanException unsupportedSqlTypeException) {
+                        log.info(unsupportedSqlTypeException.getMessage());
+                    } catch (Exception e) {
+                        // Continue to generate rollback plan for the next sql
+                        log.warn("Failed to generate rollback plan for sql:{}, error message:{}", sql, e.getMessage());
+                    }
+                }
+                handleRollbackResult(rollbackPlans.toString());
+                this.success = true;
+            } finally {
```

Comment:
it may not be nice code try catch nested

Reply:
These code is copied from `RollbackPlanRuntimeFlowableTask`, These code works right.